x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/chunking/SentenceBoundaryChunkingSettingsTests.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.chunking;
+
+import org.elasticsearch.common.ValidationException;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.inference.ChunkingStrategy;
+import org.elasticsearch.test.AbstractWireSerializingTestCase;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class SentenceBoundaryChunkingSettingsTests extends AbstractWireSerializingTestCase<SentenceBoundaryChunkingSettings> {
+
+    public void testMaxChunkSizeNotProvided() {
+        assertThrows(
+            ValidationException.class,
+            () -> { SentenceBoundaryChunkingSettings.fromMap(buildChunkingSettingsMap(Optional.empty())); }
+        );
+    }
+
+    public void testValidInputsProvided() {
+        int maxChunkSize = randomNonNegativeInt();
+        SentenceBoundaryChunkingSettings settings = SentenceBoundaryChunkingSettings.fromMap(
+            buildChunkingSettingsMap(Optional.of(maxChunkSize))
+        );
+
+        assertEquals(settings.getChunkingStrategy(), ChunkingStrategy.SENTENCE);
+        assertEquals(settings.maxChunkSize, maxChunkSize);
+    }
+
+    public Map<String, Object> buildChunkingSettingsMap(Optional<Integer> maxChunkSize) {
+        Map<String, Object> settingsMap = new HashMap<>();
+        settingsMap.put(ChunkingSettingsOptions.STRATEGY.toString(), ChunkingStrategy.SENTENCE.toString());
+        maxChunkSize.ifPresent(maxChunkSizeValue -> settingsMap.put(ChunkingSettingsOptions.MAX_CHUNK_SIZE.toString(), maxChunkSizeValue));
+
+        return settingsMap;
+    }
+
+    @Override
+    protected Writeable.Reader<SentenceBoundaryChunkingSettings> instanceReader() {
+        return SentenceBoundaryChunkingSettings::new;
+    }
+
+    @Override
+    protected SentenceBoundaryChunkingSettings createTestInstance() {
+        return new SentenceBoundaryChunkingSettings(randomNonNegativeInt());
+    }
+
+    @Override
+    protected SentenceBoundaryChunkingSettings mutateInstance(SentenceBoundaryChunkingSettings instance) throws IOException {
+        var chunkSize = instance.maxChunkSize;
+        while (chunkSize == instance.maxChunkSize) {
+            chunkSize = randomNonNegativeInt();
```

Comment:
You can replace the while loop with the more readable `randomValueOtherThan()` function

```suggestion
           var chunkSize = randomValueOtherThan(instance.maxChunkSize, ESTestCase::randomNonNegativeInt());
```

Reply:
Good catch! I was looking for just this randomizer function and must have missed it. I'll update this.